File: solved.ac/silver/5/수들의 합 5.cpp
#include <iostream>
using namespace std;

int main(void){
    int n; cin >> n;
    int cnt = 0;
    for (size_t i = 1; i < n+1; i++)
    {
        int sum = 0;
        for (size_t j = i; j < n+1; j++)
        {
            sum += j;
            if(sum >= n){
                cnt += (sum==n);
                break;
            }
        }        
    }
    cout << cnt;    
}

File: solved.ac/silver/5/수들의 합 5.py
n = int(input())

start = end = s = cnt = 1

while start != n:
    print(start,end,s)
    if s == n:
        end += 1
        cnt += 1
        s += end
    elif s > n:
        s -= start
        start += 1
    else: 
        end += 1
        s += end

print(cnt)